src/System.Net.Http/src/System/Net/Http/Unix/CurlResponseMessage.cs

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Threading;
+
+namespace System.Net.Http
+{
+    internal sealed class CurlResponseMessage : HttpResponseMessage
+    {
+        private volatile bool _disposed = false;
+        private readonly CurlResponseStream _responseStream = new CurlResponseStream();
+        private readonly ManualResetEvent _headersReceivedEvent = new ManualResetEvent(false);
+
+        #region Properties
+
+        internal ReaderWriterStream ContentStream
+        {
+            get
+            {
+                return _responseStream.InnerStream;
+            }
+        }
+
+        #endregion
+
+        internal CurlResponseMessage(HttpRequestMessage request)
+        {
+            RequestMessage = request;
+            Content = new StreamContent(_responseStream);
+        }
+
+        internal void WaitForHeaders(CancellationToken cancellationToken)
+        {
+            CheckDisposed();
+            var events = new WaitHandle[] {_headersReceivedEvent, cancellationToken.WaitHandle};
+            WaitHandle.WaitAny(events, Timeout.InfiniteTimeSpan);
+        }
+
+        internal void SignalHeadersReceived()
+        {
+            if (!_disposed)
+            {
+                _headersReceivedEvent.Set();
+#if DEBUG
+                if (Logging.On)
+                {
+                    Logging.PrintInfo(Logging.Http, this,
+                        "SignalHeadersReceived: No more headers for " + Logging.GetObjectLogHash(ContentStream));
+                }
+#endif
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing && !_disposed)
```

Comment:
That was an oversight.. it should have been disposed there. I have removed all this though since event is not needed any more (I forgot to remove the member _headersReceivedEvent - will do it in next iteration)